gradle/prCheck.gradle

```diff
@@ -0,0 +1,95 @@
+//
+// Validate Pull Request metadata
+//
+
+buildscript {
+    repositories {
+        mavenCentral()
+    }
+    dependencies {
+        classpath 'com.github.ricksbrown:cowsay:1.1.0'
+    }
+}
+
+import groovy.json.JsonSlurper
+import com.github.ricksbrown.cowsay.Cowsay
+
+def GOOD_PR_PROCESS = """
+How to pass this check?
+    1. Always have a Jira before you start working on a task (otherwise you cannot know what branch to start from).
+    2. Make sure the Jira has a 'Fix Version' defined. If it doesn't consult the team what is a good fix version for your change.
+    3. If you are working on a Jira with 'Fix Version' ending with '.1' you are working on a change that should go to a master branch. 
+       All other changes should go to the (current) fix release branch. Such branches are prefixed 'rel-'. 
+    4. Put the Jira to 'In Progress' state.
+    5. Make a code change. Include Jira ticket ID in message of the main commit (this is not checked). 
+    5. Submit a PR and ticket ID in PR's title, eg.: PUBDEV-4200: Adding support for Factorization Machines (this is checked).
+"""
+
+task checkPullRequest {
+    group = "Validation"
+    description = "Validates that PR has the right metadata in the title and is created against the correct branch"
+    doLast {
+        def errors = checkPull(System.getenv("CHANGE_ID"))
+        def allGood = errors.isEmpty()
+        def message = allGood ? "No issues found! Thank you for being awesome" : "Baa! Your PR isn't quite right"
+        def animal = allGood ? "cow" : "sheep" // as in "black sheep"
+        new File(project.buildDir, "prCheckReport.log").withWriter {w ->
+            def cowargs = ["-f", animal, message] as String[]
+            w.println(Cowsay.say(cowargs))
+            errors.forEach { e -> w.println("==>> $e") }
+            w.println()
+            w.println(GOOD_PR_PROCESS)
+        }
+        if (!errors.isEmpty()) {
+            throw new GradleException(errors[0])
+        }
+    }
+}
+
+def checkPull(changeId) {
+    def jsonSlurper = new JsonSlurper()
+    def pullDetails = jsonSlurper.parseText(new URL("https://api.github.com/repos/h2oai/h2o-3/pulls/$changeId").getText())
+    def title = pullDetails.title
+    println ""
+    println "Checking pull request ($changeId): $title"
+    println ""
+    def hexdevMatcher = title =~ /.*HEXDEV-[0-9]{4,5}.*/
+    def itaMatcher = title =~ /.*ITA-[0-9]{3,5}.*/
+    if (hexdevMatcher.matches() || itaMatcher.matches()) {
+        println "HEXDEV and ITA jiras are not validated"
```

Comment:
Any hint for developer how to fix it? Or where to seek solution.

Reply:
This doesn't produce an error, they are not just validated because they would require a private access to JIRA and they are not really part of releases so all good.